doc/adr/0009-time-graph-search-functionality.md

```diff
@@ -0,0 +1,133 @@
+
+A suggestion is to rename "regex_map_filters" to "filter_query_parameters". "filter_query_parameters" takes a "filter_expressions_map", which contains the search expressions as shown in "regex_map_filters". It would also include an optional "is_full_search" boolean. The following example does the same query given above but uses "filter_query_parameters" instead:
+```
+{
+  "parameters": {
+    "requested_timerange": {
+      "start": 111111111,
+      "end": 222222222,
+      "nbTimes": 1920
+    },
+    "requested_items": [
+      1,
+      2
+    ],
+    "filter_query_parameters": {
+	  "is_full_search": false,
+	  "filter_expressions_map": {    
+        "1": ["openat"],	   
+	  }
+    },
+  }
+}
+```
+Here are the following patches for implementing time graph filter capability through "filter_query_parameters":
+-   Update query parameters for '/states' in Trace Compass Incubator: [https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/196610](https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/196610)
+-   Updates to swagger documentation: [https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/196592](https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/196592)
+-   Updates to the TSP documentation: [https://github.com/eclipse-cdt-cloud/trace-server-protocol/pull/84](https://github.com/eclipse-cdt-cloud/trace-server-protocol/pull/84)
+
+In the previous examples we noticed that the array of search expressions takes a key value of "1” (means DIMMED) or “4” (means EXCLUDED). We should also discuss the possibility of explicitly mentioning "DIMMED" or "EXCLUDED".
+
+We also need to take a look at the response for '/states' when we add search/filter expressions in our request. The response body would still be a list of time graph entry states for each row. Every state has a bit-mask called "tags", which helps indicate whether it matched the search/filter expression or not. For example, if we ask for a 'search' expression the states that don't match it would be dimmed, and the "tags" will have the corresponding bit set:
+```
+{
+   "states": [
+     {
+       "tags": 1,
+       "start": 0,
+       "style": {
+         "parentKey": "string",
+         "values": {
+           "property1": {},
+           "property2": {}
+         }
+       },
+       "label": "string",
+       "end": 0
+     }
+   ]
+}
+```
+- When "tags" is set to 1 it is interpreted by the client as setting a dimmed opacity to the state. Similarly, when it is set to 4 it would be considered as being excluded. As previously mentioned, it could be a better alernative to have "tags" correspond explicitly to either "DIMMED" or "EXCLUDED".
+
+
```

Comment:
Can add a comment about the filter DSL? Mabye under future enhancements. Basically, the first implementation will pass the string to the back-end and it will apply it. There is no feedback if the syntax was correct. 

However, there is a possibility to create LSP server (Language Server Protocol) in the back-end and front-end will use it to do syntax validation. Trace Compass incubator already has such a server that we can integrate and have an example implementation.

Reply:
Done